container/sagemaker/python_service.py

```diff
@@ -0,0 +1,151 @@
+# Copyrasight 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import logging
+import re
+from collections import namedtuple
+
+import falcon
+import requests
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger(__name__)
+
+TFS_BASE_URI = 'http://localhost:9001/v1/models/'
+TFS_DEFAULT_MODEL = 'hal_plus_three'
+TFS_DEFAULT_METHOD = 'predict'
+DEFAULT_CONTENT_TYPE = 'application/json'
+DEFAULT_ACCEPT_HEADER = 'application/json'
+
+Context = namedtuple('Context',
+                     'model_name, model_version, method, rest_uri, grpc_uri, '
+                     'custom_attributes, request_content_type, accept_header')
+
+
+class InvocationResource(object):
+
+    def __init__(self):
+        self.handler = None
+        self.input_handler = None
+        self.output_handler = None
+        self._import_handlers()
+
+    def on_post(self, req, res):
+        context, data = self._parse_http_request(req)
+
+        if self.handler:
+            log.warning('Using handler, input_handler and output_handler will be ignored.')
+            log.info('handler type: ' + str(type(self.handler)))
+            res.status = falcon.HTTP_200
+            res.body = self.handler(data, context)
+        elif self.input_handler and self.output_handler:
```

Comment:
do we support default input/output handlers? or are we only supporting inference scripts that have (handler | (input_handler, output_handler)) functions? 

here, it look like we will fill in a default implemetation for a missing handler, but the example/test code kind of implied we might.

Reply:
We don't have default implementation of handlers, the inference_example.py is for test purpose. And the code here should be looking for 'inference.py' instead of 'inference_example.py', I refactored the name of the custom code before commit so it makes people confused.